features/eolearn/features/feature_manipulation.py

```diff
@@ -29,27 +29,29 @@
 _T = TypeVar("_T", np.ndarray, dt.datetime)
 
 
-class SimpleFilterTask(EOTask):
+class SimpleFilterTask(EOTask, Generic[_T]):
```

Comment:
wouldnt it be possible to omit all the generic stuff by using 
```python
FilterFuncType = Union[Callable[[np.ndarray], bool], Callable[[dt.datetime], bool]]
```
While i'm a big fan of using advanced type theory, I'm worried that this migh class with future improvements to EOTasks since they would need to be generics (and i'm not sure how generics clash in python).

Reply:
Ah, too bad. Ok, I removed `Generic` and `TypeVar` and used the normal union. Then I had to set a type `Iterable` to the parameter in `_get_filtered_indices` method and mypy is also fine with that.